server/src/main/java/ru/hse/server/repository/ChapterRepository.java

```diff
@@ -0,0 +1,101 @@
+package ru.hse.server.repository;
+
+import jakarta.annotation.Nonnull;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+import ru.hse.database.dao.DaoChapter;
+import ru.hse.database.entities.Chapter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class ChapterRepository implements CrudRepository<Chapter, Long> {
+
+    @Override
+    @Nonnull
+    public <S extends Chapter> S save(@Nonnull S chapter) {
+        DaoChapter.createOrUpdateChapter(chapter);
+        return chapter;
+    }
+
+    @Override
+    @Nonnull
+    public <S extends Chapter> Iterable<S> saveAll(Iterable<S> chapters) {
+        List<S> list = new ArrayList<>();
+        for (var chapter : chapters) {
+            list.add(save(chapter));
+        }
+        return list;
+    }
+
+    @Override
+    @Nonnull
+    public Optional<Chapter> findById(@Nonnull Long id) {
+        var chapter = DaoChapter.getChapterById(id);
+        if (chapter == null) {
+            return Optional.empty();
```

Comment:
Doesn't creating an Optional from null return an empty?

Reply:
Optional.of(null) throws NullPointerException